flink-runtime/src/test/java/org/apache/flink/runtime/metrics/TimerGaugeTest.java

```diff
@@ -82,5 +93,12 @@ public void testGetWithoutUpdate() {
         gauge.markEnd();
 
         assertThat(gauge.getValue(), is(0L));
+        assertThat(gauge.getMaxSingleMeasurement(), is(0L));
```

Comment:
I see that all tests check that MaxSingleMeasurement is able to grow but I don't see that we check it is also able to be decreased?
```
gauge.markStart();
clock.advanceTime(SLEEP, TimeUnit.MILLISECONDS);
gauge.markEnd();
gauge.update();
assertThat(gauge.getMaxSingleMeasurement(), is(SLEEP));

gauge.markStart();
clock.advanceTime(SLEEP/2, TimeUnit.MILLISECONDS);
gauge.markEnd();
gauge.update();
assertThat(gauge.getMaxSingleMeasurement(), is(SLEEP/2));
```

Reply:
I've added it in the first test case.